From 7f7069779855f255042f815c0ff2e90e8f29536e Mon Sep 17 00:00:00 2001
From: svenate <salini.venate@intel.com>
Date: Tue, 24 Dec 2024 05:30:19 +0000
Subject: [PATCH] Use private drm_hwcomposer instead of external 
 drm_hwcomposer.

Rename lib names to libxxx_orig, so that we use
vendor/intel/external/drm-hwcomposer instead of external
drm_hwcomposer.

Signed-off-by: svenate <salini.venate@intel.com>
---
 Android.bp       | 58 ++++++++++++++++++++++++------------------------
 tests/Android.bp | 20 ++++++++---------
 2 files changed, 39 insertions(+), 39 deletions(-)

diff --git a/Android.bp b/Android.bp
index 7d014dc..bbdf90e 100644
--- a/Android.bp
+++ b/Android.bp
@@ -30,7 +30,7 @@ license {
 }
 
 cc_library_headers {
-    name: "drm_hwcomposer_headers",
+    name: "drm_hwcomposer_headers_orig",
     vendor: true,
     export_include_dirs: ["."],
 }
@@ -39,7 +39,7 @@ cc_library_headers {
 // hwcomposer.drm.so
 // =====================
 cc_defaults {
-    name: "hwcomposer.drm_defaults",
+    name: "hwcomposer.drm_defaults_orig",
 
     shared_libs: [
         "android.hardware.graphics.composer@2.1-resources",
@@ -59,7 +59,7 @@ cc_defaults {
     ],
 
     header_libs: [
-        "drm_hwcomposer_headers",
+        "drm_hwcomposer_headers_orig",
     ],
 
     cflags: [
@@ -77,12 +77,12 @@ cc_defaults {
 }
 
 filegroup {
-    name: "drm_hwcomposer_fd",
+    name: "drm_hwcomposer_fd_orig",
     srcs: ["utils/fd.cpp"],
 }
 
 filegroup {
-    name: "drm_hwcomposer_common",
+    name: "drm_hwcomposer_common_orig",
     srcs: [
         "bufferinfo/BufferInfoGetter.cpp",
         "bufferinfo/BufferInfoMapperMetadata.cpp",
@@ -120,7 +120,7 @@ filegroup {
 }
 
 filegroup {
-    name: "drm_hwcomposer_hwc3",
+    name: "drm_hwcomposer_hwc3_orig",
     srcs: [
         "hwc3/Composer.cpp",
         "hwc3/ComposerClient.cpp",
@@ -131,21 +131,21 @@ filegroup {
 }
 
 filegroup {
-    name: "drm_hwcomposer_service",
+    name: "drm_hwcomposer_service_orig",
     srcs: [
         "hwc3/service.cpp",
     ],
 }
 
 filegroup {
-    name: "drm_hwcomposer_init_rc",
+    name: "drm_hwcomposer_init_rc_orig",
     srcs: [
         "hwc3/hwc3-drm.rc",
     ],
 }
 
 filegroup {
-    name: "drm_hwcomposer_vintf_manifest",
+    name: "drm_hwcomposer_vintf_manifest_orig",
     srcs: [
         "hwc3/hwc3-drm.xml",
     ],
@@ -153,42 +153,42 @@ filegroup {
 
 // Kept only for compatibility with older Android version. Please do not use!
 cc_library_static {
-    name: "drm_hwcomposer",
-    defaults: ["hwcomposer.drm_defaults"],
-    srcs: [":drm_hwcomposer_common"],
+    name: "drm_hwcomposer_orig",
+    defaults: ["hwcomposer.drm_defaults_orig"],
+    srcs: [":drm_hwcomposer_common_orig"],
 }
 
 cc_library_shared {
-    name: "hwcomposer.drm",
-    defaults: ["hwcomposer.drm_defaults"],
+    name: "hwcomposer.drm_orig",
+    defaults: ["hwcomposer.drm_defaults_orig"],
     srcs: [
-        ":drm_hwcomposer_common",
+        ":drm_hwcomposer_common_orig",
         "bufferinfo/legacy/BufferInfoLibdrm.cpp",
     ],
     cflags: ["-DUSE_IMAPPER4_METADATA_API"],
 }
 
 cc_library_shared {
-    name: "hwcomposer.drm_minigbm",
-    defaults: ["hwcomposer.drm_defaults"],
+    name: "hwcomposer.drm_minigbm_orig",
+    defaults: ["hwcomposer.drm_defaults_orig"],
     srcs: [
-        ":drm_hwcomposer_common",
+        ":drm_hwcomposer_common_orig",
         "bufferinfo/legacy/BufferInfoMinigbm.cpp",
     ],
 }
 
 cc_binary {
-    name: "android.hardware.composer.hwc3-service.drm",
+    name: "android.hardware.composer.hwc3-service.drm_orig",
 
     srcs: [
-        ":drm_hwcomposer_hwc3",
-        ":drm_hwcomposer_service",
-        ":drm_hwcomposer_common",
+        ":drm_hwcomposer_hwc3_orig",
+        ":drm_hwcomposer_service_orig",
+        ":drm_hwcomposer_common_orig",
         "bufferinfo/legacy/BufferInfoLibdrm.cpp",
     ],
 
     defaults: [
-        "hwcomposer.drm_defaults",
+        "hwcomposer.drm_defaults_orig",
     ],
 
     shared_libs: [
@@ -215,30 +215,30 @@ cc_binary {
     relative_install_path: "hw",
     vendor: true,
 
-    vintf_fragments: [":drm_hwcomposer_vintf_manifest"],
-    init_rc: [":drm_hwcomposer_init_rc"],
+    vintf_fragments: [":drm_hwcomposer_vintf_manifest_orig"],
+    init_rc: [":drm_hwcomposer_init_rc_orig"],
 }
 
 // Used by hwcomposer.drm_imagination
 filegroup {
-    name: "drm_hwcomposer_platformimagination",
+    name: "drm_hwcomposer_platformimagination_orig",
     srcs: ["bufferinfo/legacy/BufferInfoImagination.cpp"],
 }
 
 // Used by hwcomposer.drm_hikey and hwcomposer.drm_hikey960
 filegroup {
-    name: "drm_hwcomposer_platformhisi",
+    name: "drm_hwcomposer_platformhisi_orig",
     srcs: ["bufferinfo/legacy/BufferInfoMaliHisi.cpp"],
 }
 
 // Used by hwcomposer.drm_meson
 filegroup {
-    name: "drm_hwcomposer_platformmeson",
+    name: "drm_hwcomposer_platformmeson_orig",
     srcs: ["bufferinfo/legacy/BufferInfoMaliMeson.cpp"],
 }
 
 // Used by hwcomposer.drm_mediatek
 filegroup {
-    name: "drm_hwcomposer_platformmediatek",
+    name: "drm_hwcomposer_platformmediatek_orig",
     srcs: ["bufferinfo/legacy/BufferInfoMaliMediatek.cpp"],
 }
diff --git a/tests/Android.bp b/tests/Android.bp
index 43fd3fa..178838c 100644
--- a/tests/Android.bp
+++ b/tests/Android.bp
@@ -1,13 +1,13 @@
 cc_library_shared {
-    name: "hwcomposer.filegroups_build_test",
-    defaults: ["hwcomposer.drm_defaults"],
+    name: "hwcomposer.filegroups_build_test_orig",
+    defaults: ["hwcomposer.drm_defaults_orig"],
 
     srcs: [
-        ":drm_hwcomposer_common",
-        ":drm_hwcomposer_platformhisi",
-        ":drm_hwcomposer_platformimagination",
-        ":drm_hwcomposer_platformmediatek",
-        ":drm_hwcomposer_platformmeson",
+        ":drm_hwcomposer_common_orig",
+        ":drm_hwcomposer_platformhisi_orig",
+        ":drm_hwcomposer_platformimagination_orig",
+        ":drm_hwcomposer_platformmediatek_orig",
+        ":drm_hwcomposer_platformmeson_orig",
     ],
 
     local_include_dirs: [
@@ -30,16 +30,16 @@ package {
 
 // Tool for listening and dumping uevents
 cc_test {
-    name: "hwc-drm-uevent-print",
+    name: "hwc-drm-uevent-print_orig",
 
     srcs: [
-        ":drm_hwcomposer_fd",
+        ":drm_hwcomposer_fd_orig",
         "uevent_print.cpp",
     ],
 
     vendor: true,
     header_libs: [
-        "drm_hwcomposer_headers",
+        "drm_hwcomposer_headers_orig",
         "libhardware_headers",
     ],
     shared_libs: ["liblog"],
-- 
2.34.1

